### ISSUE TITLE
Feature request tap_action: custom with own javascript

### ISSUE THREAD

**Issue by sVnsation** [Nerwyn/service-call-tile-feature#50]
Hello Nerwyn,

first of all: thanks for this great addon to the tile cards.

It would be nice to call **custom tap_actions** with own **javascript**, like it's possible with [custom:button-card](https://github.com/custom-cards/button-card).
With this you can have multiple actions at once, for example call a service and navigate to a new page.

With custom:button-card the yaml-code works as follows:
```
tap_action:
  action: custom
  custom: |
    [[[
    hass.callService("script", "camera", {});
    window.history.pushState(null, "", "/lovelace/camera");
    window.dispatchEvent(new CustomEvent("location-changed"));
    ]]]
```
Looking forward to the project getting even better with this addition.

Best regards

**Comment by Nerwyn** [comment]
I'm having trouble finding any reference to a `custom` action in the custom:button-card documentation or code. Are you sure it isn't from a different custom card?

I should be able to implement this using [eval](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval).

**Comment by sVnsation** [comment]
Oh, it seems to be an undocumented function. When searching for how to call multiple functions, I only found a post about it in the home assitant community forum. But here on github there is also something written about it:

https://github.com/custom-cards/button-card/issues/314

**Comment by Nerwyn** [comment]
Try the latest 4.1.0 alpha. It's the action Evaluate JS in the configuration UI.

**Comment by sVnsation** [comment]
Tried it. Navigation with `window.`works.
But log says `hass is not defined eval` when I tried to execute a command from here https://developers.home-assistant.io/docs/frontend/data/

**Comment by Nerwyn** [comment]
Use `this.hass`. The code is evaluated from within the context of the custom feature, of which `hass` is a property/member.

**Comment by sVnsation** [comment]
Works perfect.

**Comment by Nerwyn** [comment]
[Released 4.1.0!](https://github.com/Nerwyn/service-call-tile-feature/releases/tag/4.1.0)
